honestcash-v2/src/app/store/wallet/wallet.reducer.ts

```diff
@@ -1,29 +0,0 @@
-
-import { WalletActionTypes, All } from './wallet.actions';
-import { State, initialState } from './wallet.state';
-import { WalletUtils } from '../../shared/lib/WalletUtils';
-
-export function reducer(state = initialState, action: All): State {
-  switch (action.type) {
-    case WalletActionTypes.WALLET_SETUP: {
```

Comment:
this is pure.

Reply:
Most certainly! Reducers should be pure functions, so the logic must be pure as well. This means there should be zero side effects. Side effects would include (but not limited to):
```
Database requests/stores
File IO
REST/async calls
Global or external mutations
Data mutations of any sort
```
https://stackoverflow.com/a/47228099